mm2src/coins/tendermint/tendermint_token.rs

```diff
@@ -757,18 +703,18 @@ impl MmCoin for TendermintToken {
             let fee = Fee::from_amount_and_gas(fee_amount, gas_limit);
 
             let account_info = platform.account_info(&account_id).await?;
-            let tx_raw = platform
-                .any_to_signed_raw_tx(&priv_key, account_info, msg_send, fee, timeout_height, memo.clone())
-                .map_to_mm(|e| WithdrawError::InternalError(e.to_string()))?;
 
-            let tx_bytes = tx_raw
-                .to_bytes()
+            let tx = platform
+                .any_to_transaction_data(maybe_pk, msg_send, account_info, fee, timeout_height, memo.clone())
                 .map_to_mm(|e| WithdrawError::InternalError(e.to_string()))?;
 
-            let hash = sha256(&tx_bytes);
+            let internal_id = {
+                let hex_vec = tx.tx_hex().cloned().unwrap_or_default().to_vec();
+                sha256(&hex_vec).to_vec().into()
+            };
+
```

Comment:
does this mean we'll have duplicate `internal_id` if we end up in `unwrap_or_default` multiple times?

Reply:
Where exactly `internal_id` is duplicated? We don't save it in anywhere AFAIK.

Reply:
never mind...was reading something else